From: Gordon Bos <gordon@bosvangennip.nl>
Date: Wed, 14 Oct 2020 12:15:44 +0200
Subject: Bindist conflict on newly enabled protocols

net-wireless/hostapd-2.9-r3 enables three EAP authentication methods
requiring library functions in ssl for which no permision is granted
to distribute these as a binary.

These methods are:

* Dragonfly (CONFIG_SAE)
* Opportunistic Wireless Encryption (CONFIG_OWE)
* Device Provisioning Protocol (CONFIG_DPP)

This patch changes the hostapd makefile to conditionally disregard
these additional protocols, based on the `internal-tls` USE flag
that is enabled as part of the bindist restrictions on the live USB 


--- a/hostapd/Makefile	2019-08-07 15:25:25.000000000 +0200
+++ b/hostapd/Makefile	2020-10-14 11:26:24.071085457 +0200
@@ -307,6 +307,7 @@
 OBJS += ../src/ap/eth_p_oui.o
 endif
 
+ifndef CONFIG_TLS
 ifdef CONFIG_SAE
 CFLAGS += -DCONFIG_SAE
 OBJS += ../src/common/sae.o
@@ -315,7 +316,9 @@
 NEED_AP_MLME=y
 NEED_DRAGONFLY=y
 endif
+endif
 
+ifndef CONFIG_TLS
 ifdef CONFIG_OWE
 CFLAGS += -DCONFIG_OWE
 NEED_ECC=y
@@ -326,6 +329,7 @@
 NEED_SHA384=y
 NEED_SHA512=y
 endif
+endif
 
 ifdef CONFIG_AIRTIME_POLICY
 CFLAGS += -DCONFIG_AIRTIME_POLICY
@@ -590,6 +590,7 @@
 
 endif
 
+ifndef CONFIG_TLS
 ifdef CONFIG_DPP
 CFLAGS += -DCONFIG_DPP
 OBJS += ../src/common/dpp.o
@@ -609,6 +610,7 @@
 CFLAGS += -DCONFIG_DPP2
 endif
 endif
+endif
 
 ifdef CONFIG_EAP_IKEV2
 CFLAGS += -DEAP_SERVER_IKEV2

